### ISSUE TITLE
Is scaffold supported in PromptCache?

### ISSUE THREAD

**Issue by dequeueing** [yale-sys/prompt-cache#20]
Hi. I have read your great paper and I feel quite inspired. In the paper you mention that: 

> One way to remove the masking effect is to use a method we refer to as scaffolding. At the cost of additional memory, we allow users to specify "scaffolds", which are sets of prompt modules that are encoded together to share the attention span, in addition to their individual attention states.

I wonder is scaffold implemented in PromptCache? I could not find anything related to it in the repo.

Also, I notice that in demo.py, all the modules' attention are calculated together, not individually:

https://github.com/yale-sys/prompt-cache/blob/9027040d4b4e3fce911b060590dd4a71783b12b9/promptcache/cache_engine.py#L240C1-L248C18
```python
        batch_token_ids_padded, attn_mask = pad_batch(batch_token_ids, self.lm.eos_token_id)
        batch_position_ids_padded, _ = pad_batch(batch_position_ids, 0)

        d_output = self.lm(
            input_ids=torch.tensor(batch_token_ids_padded, device=self.lm.device, dtype=torch.long),
            position_ids=torch.tensor(batch_position_ids_padded, device=self.lm.device, dtype=torch.long),
            attention_mask=torch.tensor(attn_mask, device=self.lm.device, dtype=torch.float16),
            use_cache=True
        )
```

I think these modules are actually in a scaffold, as mentioned in the paper. I hope I could get your reply. Thank you very much!
